resource/graph_node.go

```diff
@@ -32,14 +50,18 @@ type GraphNode struct {
 	current                   Resource
 	currentModel              Model
 	config                    Config
-	needsReconfigure          bool
 	lastReconfigured          *time.Time
 	lastErr                   error
-	markedForRemoval          bool
 	unresolvedDependencies    []string
 	needsDependencyResolution bool
 
 	logger logging.Logger
+
+	// state stores the current lifecycle state for a resource node
+	state NodeState
+	// transitionedAt stores the timestamp of when resource entered it's current
+	// lifecycle state.
+	transitionedAt time.Time
```

Comment:
Maybe a `*time.Time` as for `lastReconfigured`? The zero-value of a `time.Time` is something like Jan 1. 1970.